File: src/element/event.ts
import { MatchResult } from "../match";
import { Parser } from "../parser";
import { Pass } from "../pass/pass";
import { events } from "../registry";
import { StructureStatement } from "../statement/structure";
import { Element, ElementPriority, ElementType } from "./element";

export abstract class Event implements Element {
    public getPriority(): ElementPriority {
        return ElementPriority.NORMAL;
    }

    public getElementType(): ElementType {
        return ElementType.EVENT;
    }

    public abstract visit(pass: Pass, statement: StructureStatement): boolean;
    public abstract initialize();
}

File: common/src/main/kotlin/me/honkling/skriptkt/common/syntax/Event.kt
package me.honkling.skriptkt.common.syntax

import me.honkling.skriptkt.common.events.Event
import me.honkling.skriptkt.common.statement.StructureStatement

abstract class Event<T : Event>(val event: Class<T>) : Structure() {
    abstract fun check(event: T, statement: StructureStatement): Boolean
}
